### PR TITLE
Fix #283. Jira API V2 is expecting a 'name' property

### LINKED ISSUE #283: Unable to unassign issue
When trying to call `org-jira-unassign-issue` on an issue, I get an error, and see the following output in the `*Messages*` buffer:

```
[error] request--callback: peculiar error: 400

"JIRA_ERROR - see your *Messages* buffer for more details."

"JIRA_ERROR REQUEST: "

"/rest/api/2/issue/XXX-971"

(:type "PUT" :data "{\"fields\":{\"components\":[],\"priority\":{\"id\":\"10000\",\"name\":\"Default\"},\"description\":\"Ensure that Jarvis can be used to perform actions (block/unblock, bypass, add labels, etc.)\",\"assignee\":{\"id\":null},\"reporter\":{\"id\":null},\"summary\":\"Jarvis actions\",\"issuetype\":{\"id\":\"7\",\"name\":\"Story\"}}}")

"JIRA_ERROR RESPONSE: "

((errorMessages . []) (errors (assignee . "expected Object containing a 'name' property")))
```

I'm running the latest org-jira as of today, via Melpa.

### PR DESCRIPTION
**Pull request by jeromeag**
This fix is needed for Tickets updates with a recent Jira server.
(naive copy/paste from [pjgarcia](https://github.com/pjgarcia) comment in #283) 
I don't know if this fix is working also on legacy Jira servers.
Maybe it is needed to test Jira API version to implement a better fix.

### CODE REVIEW

File: jiralib.el [hunk 1]
```diff
@@ -942,10 +942,10 @@ Return nil if the field is not found"
         return (cdr (assoc 'displayName user))))
 
 (defun jiralib-get-user-account-id (project full-name)
```
Comment (ahungry) [inline jiralib.el:944]: I didn't test it - the change looks good, but leaves the function name as a misnomer - if it's no longer returning an account-id, it should be changed (or left alone, and a new one added) called `jiralib-get-user-name` or similar.

**Review by ahungry — CHANGES_REQUESTED** [review]
See comment

**Comment by ahungry** [comment]
You mention api v2, which I assume based on a comment here: https://github.com/ahungry/org-jira/issues/305 is the API version tied to your on prem jira version.

Work was done awhile ago to ensure account-id worked instead of name, for various things (like assignee/reporter id) - this change seems to go in the opposite direction, and I'm wondering how compatible it is with jira cloud/v3 api - can someone with cloud access please test the changes out?

I checked their docs now, and the REST docs are pretty unreadable atm (supposed to be dynamic or something, and instead of listing out the params, they have "meta" endpoints to do it now) - the couple searches I did for account-id now seem to imply account-id and name are interchangeable things.

I know cloud jira descriptions quit support name type tagging like `[~mcarter]` some time ago, in favor of long uuid (account-id) values.

**Comment by jeromeag** [comment]
At first, I mentioned api v2 based on a comment here [ issue 283](https://github.com/ahungry/org-jira/issues/283#issuecomment-1047925882).
The proposed fix worked for me... and I have done a PR expecting (naively) the fix to be included the next release.

It seems that a more complete fix would be to first check the API version.
I'm afraid that my elisp/rest skills wil not be sufficient.

**Comment by ahungry** [comment]
At some point I made an org-jira-sdk abstraction to encapsulate the field values of the various REST entities (issue / comment / board) - I wonder if doing some type of dispatch/multimethod based on a user setting for "remote api version" (or maybe we can set that dynamically) would allow the code to have a linear flow, while still supporting multiple versions/hosting solutions of jira.  In the past, I had to postpone merge on some "on-prem" changes, as they didn't work (or were untested) against the cloud version.

What's the minimal acceptance test/outline of a scenario to produce the failure and subsequent fix your PR addresses?

Something like:

- File an issue with org-jira-create-issue
- Observe the assignee/reporter are failed to be set

**Comment by jeromeag** [comment]
Without this modification the **org-jira-update-issue** call was simply failing with "expected Object containing a 'name' property" Jira server error. (I googled it and felt on #283 issue event if the way to get the error is not the same).

So to reproduce:
- update an issue with org-jira-update-issue
- observe: 
> "JIRA_ERROR RESPONSE: "
> ((errorMessages . []) (errors (assignee . "expected Object containing a 'name' property")))

**Comment by ahungry** [comment]
I tested this out on the cloud - name isn't a valid piece of the assignee object - it defers to accountId - I think a couple people may be keeping their own customized fork for working with on-prem jiras - as (I believe) cloud is the most common jira option (certainly the most up to date one) I want to keep that as the main target, so I'll have to pass on this PR for now - thank you though - it would be a good reference if I plan to add non-cloud support targets.

**Pull request closed without merging**